crates/provider/src/fillers/mod.rs

```diff
@@ -200,6 +200,19 @@ pub trait TxFiller<N: Network = Ethereum>: Clone + Send + Sync + std::fmt::Debug
         tx: SendableTx<N>,
     ) -> impl_future!(<Output = TransportResult<SendableTx<N>>>);
 
+    /// Fills in the transaction request and try to convert it to an envelope.
+    fn fill_envelope(
+        &self,
+        fillable: Self::Fillable,
+        tx: SendableTx<N>,
+    ) -> impl_future!(<Output = TransportResult<N::TxEnvelope>>) {
+        async move {
+            let tx = self.fill(fillable, tx).await?;
+            let envelope = tx.try_into_envelope().map_err(/* Not sure where to create the new error type */)?;
```

Comment:
yeah we can either introduce a new dedicated error type (whcih I think would be fine here)

or we simply type erase this error to https://github.com/alloy-rs/alloy/blob/833f9a22ca7c30079d64554d435561877c15267a/crates/transport/src/error.rs#L57-L57

both approaches are fine here

I'd prefer the additional error enum with two variants